polkadot/runtime/westend/src/lib.rs

```diff
@@ -1563,6 +1569,144 @@ pub type TxExtension = (
 	pallet_transaction_payment::ChargeTransactionPayment<Runtime>,
 );
 
+#[test]
+fn free_transaction_extension_test() {
+	sp_io::TestExternalities::default().execute_with(|| {
+		use frame_support::{
+			dispatch::{DispatchInfo, RawOrigin},
+			traits::BuildGenesisConfig,
+		};
+		use keyring::AccountKeyring;
+		use sp_io::hashing::blake2_256;
+		use sp_runtime::{
+			traits::{DispatchTransaction, TransactionExtensionBase},
+			MultiSignature,
+		};
+
+		// The part of `TxExtension` that has to be provided and signed by user who wants
+		// the transaciton fee to be sponsored by someone else.
+		type BaseTxExtension = (
```

Comment:
FYI there is https://paritytech.github.io/polkadot-sdk/master/frame/runtime/types_common/type.SystemSignedExtensionsOf.html exactly to simplify such setups :)